File: thinkup-api/thinkup-api/src/main/java/com/example/thinkup_api/model/Idea.java
package com.example.thinkup_api.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Idea {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;
    private String description;
    private String category;
    private double lat;
    private double lng;
    private String author;
    private long createdAt = System.currentTimeMillis();
}

File: app/src/main/java/com/example/thinkup/model/Idea.kt
package com.example.thinkup.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "ideas")
data class Idea(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val title: String,
    val description: String,
    val category: String,
    val lat: Double,
    val lng: Double,
    val author: String,
    val createdAt: Long = System.currentTimeMillis()
)
